pkg/loki/topology_query.go

```diff
@@ -74,10 +74,24 @@ func NewTopologyQuery(cfg *Config, start, end, limit, metricFunction, metricType
 			limit:          l,
 			metricFunction: f,
 			dataField:      t,
+			fields:         getScopeFields(scope),
 		},
 	}, nil
 }
 
+func getScopeFields(scope string) string {
+	switch scope {
+	case "host":
+		return "SrcK8S_HostIP,DstK8S_HostIP"
+	case "namespace":
+		return "SrcK8S_Namespace,DstK8S_Namespace"
+	case "owner":
+		return "SrcK8S_OwnerName,SrcK8S_OwnerType,SrcK8S_Namespace,SrcK8S_HostIP,DstK8S_OwnerName,DstK8S_OwnerType,DstK8S_Namespace,DstK8S_HostIP"
```

Comment:
I think it's both easier and cleaner to just remove HostIP here

Reply:
what about dynamically add them according to groups ? So we will never have a duplicate without grouping / inside each group

Reply:
you mean for the special case when scope == owner and grouping by node?

Reply:
yes, check https://github.com/netobserv/network-observability-console-plugin/pull/131/files#diff-993c47cff38972b388f389064fe50dd3db1feb60eb3d811cbc17727968f2d23fR85